crates/router/src/connector/trustpay.rs

```diff
@@ -0,0 +1,637 @@
+        &self,
+        auth_type: &types::ConnectorAuthType,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let auth = trustpay::TrustpayAuthType::try_from(auth_type)
+            .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+        Ok(vec![(headers::X_API_KEY.to_string(), auth.api_key)])
+    }
+
+    fn build_error_response(
+        &self,
+        res: Response,
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        logger::debug!(trustpay_error_response=?res);
+        let response: trustpay::TrustpayErrorResponse = res
+            .response
+            .parse_struct("trustpay ErrorResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        let default_error = trustpay::Errors {
+            code: 0,
+            description: consts::NO_ERROR_CODE.to_string(),
+        };
+        Ok(ErrorResponse {
+            status_code: res.status_code,
+            code: response.status.to_string(),
+            message: format!("{:?}", response.errors.first().unwrap_or(&default_error)),
+            reason: None,
+        })
+    }
+}
+
+fn get_trustpay_headers(
+    payment_method: storage_models::enums::PaymentMethod,
+    content_type: String,
+    auth_header: Vec<(String, String)>,
+    access_token: Option<&types::AccessToken>,
+) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+    match payment_method {
+        storage_models::enums::PaymentMethod::BankRedirect => {
+            let token = access_token.ok_or(errors::ConnectorError::FailedToObtainAuthType)?;
+            Ok(vec![
+                (
+                    headers::CONTENT_TYPE.to_string(),
+                    "application/json".to_owned(),
+                ),
+                (
+                    headers::AUTHORIZATION.to_string(),
+                    format!("Bearer {}", token.token),
+                ),
+            ])
+        }
+        _ => {
+            let mut header = vec![(headers::CONTENT_TYPE.to_string(), content_type)];
+            let mut api_key = auth_header;
+            header.append(&mut api_key);
+            Ok(header)
+        }
+    }
+}
+
+impl api::Payment for Trustpay {}
+
+impl api::PreVerify for Trustpay {}
+impl ConnectorIntegration<api::Verify, types::VerifyRequestData, types::PaymentsResponseData>
+    for Trustpay
+{
+}
+
+impl api::PaymentVoid for Trustpay {}
+
+impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>
+    for Trustpay
+{
+}
+
+impl api::ConnectorAccessToken for Trustpay {}
+
+impl ConnectorIntegration<api::AccessTokenAuth, types::AccessTokenRequestData, types::AccessToken>
+    for Trustpay
+{
+    fn get_url(
+        &self,
+        _req: &types::RefreshTokenRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok("https://aapi.trustpay.eu/api/oauth2/token".to_owned())
```

Comment:
Don't hard code like this, if this is production url then write a logic to find then testing or production and for production use it from env only and if test hard code the url